planner/core/exhaust_physical_plans.go

```diff
@@ -2751,11 +2751,48 @@ func (la *LogicalAggregation) exhaustPhysicalPlans(prop *property.PhysicalProper
 }
 
 func (p *LogicalSelection) exhaustPhysicalPlans(prop *property.PhysicalProperty) ([]PhysicalPlan, bool, error) {
+	newProps := make([]*property.PhysicalProperty, 0, 3)
 	childProp := prop.CloneEssentialFields()
-	sel := PhysicalSelection{
-		Conditions: p.Conditions,
-	}.Init(p.ctx, p.stats.ScaleByExpectCnt(prop.ExpectedCnt), p.blockOffset, childProp)
-	return []PhysicalPlan{sel}, true, nil
+	newProps = append(newProps, childProp)
+
+	if !expression.ContainVirtualColumn(p.Conditions) {
+		mppAllowed := p.SCtx().GetSessionVars().IsMPPAllowed()
+
+		if prop.TaskTp != property.MppTaskType &&
+			mppAllowed && p.canPushDown(kv.TiFlash) {
+			childPropMpp := prop.CloneEssentialFields()
+			childPropMpp.TaskTp = property.MppTaskType
+			newProps = append(newProps, childPropMpp)
+		}
+
+		// push down to TiKV or cop TiFlash
+		if prop.TaskTp != property.CopSingleReadTaskType &&
+			(p.canPushDown(kv.TiKV) ||
+				!mppAllowed && p.canPushDown(kv.TiFlash)) {
+			childPropMpp := prop.CloneEssentialFields()
```

Comment:
Use `childPropCop` instead of `childPropMpp`?

Reply:
Updated.